### PR TITLE
[OTE-815] Update GetTierForAffiliates and MsgRegisterAffiliates to handle empty tiers

### PR DESCRIPTION
**Pull request by affanv14**
### Changelist
[Describe or list the changes made in this PR]

### Test Plan
[Describe how this PR was tested (if applicable)]

### Author/Reviewer Checklist
- [ ] If this PR has changes that result in a different app state given the same prior state and transaction list, manually add the `state-breaking` label.
- [ ] If the PR has breaking postgres changes to the indexer add the `indexer-postgres-breaking` label.
- [ ] If this PR isn't state-breaking but has changes that modify behavior in `PrepareProposal` or `ProcessProposal`, manually add the label `proposal-breaking`.
- [ ] If this PR is one of many that implement a specific feature, manually label them all `feature:[feature-name]`.
- [ ] If you wish to for mergify-bot to automatically create a PR to backport your change to a release branch, manually add the label `backport/[branch-name]`.
- [ ] Manually add any of the following labels: `refactor`, `chore`, `bug`.

## Summary by CodeRabbit

- **New Features**
	- Enhanced affiliate registration process to ensure predefined tiers are set before registration.
	- Introduced a specific error message for cases where affiliate tiers are not configured.

- **Bug Fixes**
	- Improved error handling for retrieving tier information when no tiers are available.

- **Tests**
	- Added new tests to validate behavior when no affiliate tiers are set, ensuring robust error handling and expected defaults.
	- Updated test setups to ensure affiliate tiers are initialized before executing tests, enhancing reliability.

### CODE REVIEW

File: protocol/x/affiliates/keeper/keeper.go [hunk 1]
```diff
@@ -204,6 +212,10 @@ func (k Keeper) GetTierForAffiliate(
 		return 0, 0, err
 	}
 	tiers := affiliateTiers.GetTiers()
+	// Return 0 tier if no tiers are set.
+	if len(tiers) == 0 {
+		return 0, 0, errorsmod.Wrapf(types.ErrAffiliateTiersNotSet, "affiliate tiers are not set")
```
Comment (teddyding) [inline protocol/x/affiliates/keeper/keeper.go:217]: Should we return `0, 0, nil` here to be safe? Thinking about the case where:
- Governance approves affiliate tiers
- Alice refers Bob
- Governane update affiliate tiers to empty
- No all taker trade by Bob will result in error (?)

By default returning a 0 tier and 0 fee share is not the most accurate but I believe fine practically, and also the safest. Really what we want to return is an additional variable `isAffiliateProgramActive`, but that's an overkill

File: protocol/x/affiliates/types/errors.go [hunk 2]
```diff
@@ -14,4 +14,5 @@ var (
 		ModuleName, 7, "Rev share safety violation")
 	ErrDuplicateAffiliateAddressForWhitelist = errorsmod.Register(
 		ModuleName, 8, "Duplicate affiliate address for whitelist")
+	ErrAffiliateTiersNotSet = errorsmod.Register(ModuleName, 9, "Affiliate tiers not set")
```
Comment (teddyding) [inline protocol/x/affiliates/types/errors.go:17]:
```suggestion
	ErrAffiliateTiersNotSet = errorsmod.Register(ModuleName, 9, "Affiliate tiers not set (affiliate program is not active)")
```

File: protocol/x/affiliates/keeper/keeper.go (hunk 1 above)
Comment (affanv14) [reply protocol/x/affiliates/keeper/keeper.go]: agreed

File: protocol/x/affiliates/keeper/keeper.go [hunk 3]
```diff
@@ -71,6 +71,14 @@ func (k Keeper) RegisterAffiliate(
 		return errorsmod.Wrapf(types.ErrAffiliateAlreadyExistsForReferee, "referee: %s, affiliate: %s",
 			referee, affiliateAddr)
 	}
+	affiliateTiers, err := k.GetAllAffiliateTiers(ctx)
+	if err != nil {
+		return err
+	}
+	// Return error if no tiers are set.
+	if len(affiliateTiers.GetTiers()) == 0 {
+		return errorsmod.Wrapf(types.ErrAffiliateTiersNotSet, "affiliate tiers are not set")
```
Comment (teddyding) [inline protocol/x/affiliates/keeper/keeper.go:80]: This is redundant, just return `types.ErrAffiliateTiersNotSet` directly

**Review by teddyding — APPROVED** [review]

**Review by teddyding — APPROVED** [review]

**Review by teddyding — APPROVED** [review]

**Pull request merged**